lib/toggle-quotes.coffee

```diff
@@ -13,7 +14,9 @@ toggleQuoteAtPosition = (editor, position) ->
     # This is useful for languages where changing the quotes makes the range
     # invalid and so toggling again should properly restore the valid quotes
     if range = editor.displayBuffer.bufferRangeForScopeAtPosition('.invalid.illegal', position)
-      return unless /^(".*"|'.*')$/.test(editor.getTextInBufferRange(range))
+      inner = quoteChars.split('').map((c) -> "#{c}.*#{c}").join('|')
+      regex = new RegExp("^(#{inner})$", 'g')
```

Comment:
CoffeeScript has a nice `///` option for interpolated regexes, so this could be:

```coffee
///^(#{inner})$///g
```